solutions/cses-1161.mdx

```diff
@@ -55,4 +54,39 @@ int main(){
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class StickDivision {
+	public static void main(String[] args) throws IOException {
+		BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+		StringTokenizer st = new StringTokenizer(in.readLine());
+
+		// with our algorithm the variable x will not be used at all
+		// since it is guaranteed by problem constraints that all values add up to x
+		int x = Integer.parseInt(st.nextToken());
+		int n = Integer.parseInt(st.nextToken());
+		Queue<Integer> sticks = new PriorityQueue<Integer>();
+		st = new StringTokenizer(in.readLine());
+		for (int i = 0; i < n; i++) {
+			sticks.add(Integer.valueOf(st.nextToken()));
+		}
+		in.close();
+
+		long costs = 0;
+		while (sticks.size() > 1) {
+			int a = sticks.poll(), b = sticks.poll();
```

Comment:
more explanation here?

Reply:
think it's explained in the editorial, but yeah a small comment like `// take the two smallest sticks` would help

Reply:
Added some explanations, maybe have a look if there are something unclear/grammatic issues.